### PR TITLE
Add ranch coverage for LRO

### PR DESCRIPTION
**Pull request by MaryGao**
guideline: https://github.com/microsoft/api-guidelines/blob/vNext/azure/ConsiderationsForServiceDesign.md#put-with-additional-long-running-processing

see final design here: https://gist.github.com/MaryGao/fa84b3de31735db16452c2012d2dc831

fixes https://github.com/Azure/autorest.typescript/issues/1558

### CODE REVIEW

File: packages/typespec-ts/test/integration/lroCore.spec.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+} from "./generated/lro/lroCore/src/index.js";
+import { assert } from "chai";
+describe.only("AzureLroCoreClient Rest Client", () => {
+  let client: AzureLroCoreClient;
+
+  beforeEach(() => {
+    client = AzureLroCoreClientFactory({ allowInsecureConnection: true });
+  });
+
+  it("should put LRO response", async () => {
```
Comment (MaryGao) [inline packages/typespec-ts/test/integration/lroCore.spec.ts:14]: For this PUT LRO there will be no polling status monitore operation generated.

File: packages/typespec-test/test/anomalyDetector/generated/openapi/v1.1/openapi.json [hunk 2]
```diff
@@ -291,9 +291,10 @@
                 "description": "Id of the detection result.",
                 "type": "string"
               },
-              "operation-location": {
-                "description": "Location of the detection result.",
-                "type": "string"
+              "Operation-Location": {
+                "description": "The location of an instance of ResourceOperationStatus",
+                "type": "string",
+                "format": "uri"
```
Comment (qiaozha) [inline packages/typespec-test/test/anomalyDetector/generated/openapi/v1.1/openapi.json:297]: why do we have this change ?

File: packages/autorest.typescript/src/restLevelClient/transforms/transformPaths.ts [hunk 3]
```diff
@@ -91,7 +91,9 @@ export function transformPaths(model: CodeModel): Paths {
             successStatus: gerOperationSuccessStatus(operation),
             operationName,
             annotations: {
```
Comment (MaryGao) [inline packages/autorest.typescript/src/restLevelClient/transforms/transformPaths.ts:93]: Update the name

File: packages/typespec-ts/test/integration/generated/lro/lroCore/src/pollingHelper.ts [hunk 4]
```diff
@@ -10,13 +10,51 @@ import {
...
+>(
+  client: Client,
+  initialResponse: Delete202Response | DeleteDefaultResponse,
+  options?: CreateHttpPollerOptions<TResult, OperationState<TResult>>
+): Promise<SimplePollerLike<OperationState<TResult>, TResult>>;
+export async function getLongRunningPoller<
+  TResult extends ExportLogicalResponse | ExportDefaultResponse
+>(
+  client: Client,
+  initialResponse: Export202Response | ExportDefaultResponse,
```
Comment (MaryGao) [inline packages/typespec-ts/test/integration/generated/lro/lroCore/src/pollingHelper.ts:55]: Update the overload positions

File: packages/typespec-ts/src/operationUtil.ts [hunk 5]
```diff
@@ -82,27 +85,55 @@ export function isLongRunningOperation(
...
-      for (const [_, header] of Object.entries(unit.headers!)) {
-        if (hasDecorator(header, "$pollingLocation")) {
-          return true;
-        }
-      }
-    }
+  return Boolean(getLroMetadata(program, operation.operation));
+}
+
+export function shouldGenerateLroLogicalResponse(
```
Comment (MaryGao) [inline packages/typespec-ts/src/operationUtil.ts:91]: Reduce the name confusion

**Review by qiaozha — APPROVED** [review]

**Pull request merged**